src/spectral.jl

```diff
@@ -264,3 +264,25 @@ function contract(nbt::Nonbacktracking, edgespace::Vector)
     contract!(y,nbt,edgespace)
     return y
 end
+
+"""spectral_distance(G₁, G₂ [, k])
+Compute the spectral distance between undirected graphs
+G₁ and G₂ using the top k greatest eigenvalues. If k is
+ommitted, uses k = min(|G₁|,|G₂|).
+
+For further details, please refer to:
+
+JOVANOVIC, I.; STANIC, Z., 2014. Spectral Distances of
+Graphs Based on their Different Matrix Representations
+"""
+function spectral_distance(G₁::Graph, G₂::Graph, k::Integer)
+  A₁ = adjacency_matrix(G₁)
+  A₂ = adjacency_matrix(G₂)
+
+  λ₁ = k < nv(G₁)-1 ? eigs(A₁, nev=k, which=:LR)[1] : eigvals(full(A₁))[end:-1:end-(k-1)]
+  λ₂ = k < nv(G₂)-1 ? eigs(A₂, nev=k, which=:LR)[1] : eigvals(full(A₂))[end:-1:end-(k-1)]
```

Comment:
Sorry, I cannot access that articl from home. 

In any case my point is: let's call 'd' the distance involving all the eigenvalues, and `d_k` the distance involving the first `k` eigenvalues according to some ordering. Of course,  calling `n` the number of nodes, we have `d_n=d` for any ordering of the eigenvalues. 
In the first article you linked only `d` was involved, from the brief glance I gave. If the purpose of `d_k` is to have a computationally convenient way to approximate `d`, then it is better to sort the eigs by magnitude.   On the other hand, if `d_k` has its own meaning and purpouse, we just stick with that meaning. Since I think we are in the former case, we should sort by magnitude